### PR TITLE
settings moved to gcode pkg, simplified usage of the gcode

### PR DESCRIPTION
**Pull request by l1va**

### CODE REVIEW

File: gcode/cmds.go [hunk 1]
```diff
@@ -49,87 +72,108 @@ type RotateZ struct {
...
-	Layers    []Layer
-	Index     int
-	ExtParams ExtrusionParams
+	Layers []Layer
 }
 
-func (lm LayersMoving) ToGCode(b *bytes.Buffer) {
-	eOff := 0.0
-	//reset extruder encoder
-	b.WriteString("G92 E0\n")
```
Comment (hircumg) [inline gcode/cmds.go:68]: Why you removed reseting extruder and the beginning of gcode?

File: gcode/cmds.go [hunk 2]
```diff
@@ -49,87 +72,108 @@ type RotateZ struct {
...
+		state.eOff = 0
 	}
+	return state
 }
 
-func switchFanGCode(fanOff bool, b *bytes.Buffer) {
-	if fanOff {
-		b.WriteString("M107\n")
+func switchFanGCode(state State, sett GcodeSettings, b *bytes.Buffer) State {
+	if state.layerIndex == 0 && sett.FanOffLayer1 {
```
Comment (hircumg) [inline gcode/cmds.go:119]: What if we desire to turn on fan after one layer, e.g. layerIndex == 1

File: gcode/cmds.go (hunk 1 above)
Comment (l1va) [reply gcode/cmds.go:68]: header template has initial setting and i think it is not needed. Only for resetting if it is too long (somewhere below)

File: gcode/cmds.go (hunk 2 above)
Comment (l1va) [reply gcode/cmds.go:119]: This setting is only about turn on/off fan on the first layer, if it will be another setting - it will be another code.

**Pull request merged**